### PR TITLE
feat: add support for k8s 1.29

### PR DESCRIPTION
**Pull request by aboguszewski-sumo**
Some `coredns_forward` metrics have been deprecated: 
https://github.com/dioad/coredns/blob/1918a715988f40a9d31fb386ecf814832e62cbc2/plugin/forward/README.md?plain=1#L114

so the tests did not pass on k8s 1.29

### Checklist

- [x] Changelog updated or skip changelog label added
- [X] Documentation updated
- [ ] Template tests added for new features
- [X] Integration tests added or modified for major features

### CODE REVIEW

File: tests/integration/internal/constants.go [hunk 1]
```diff
@@ -324,6 +328,22 @@ var (
 		"scrape_samples_scraped",
 		"scrape_duration_seconds",
 	}
+
+	// Some metrics might change over k8s versions
+	// It is assumed that
```
Comment (swiatekm-sumo) [inline tests/integration/internal/constants.go:333]: What is assumed? :smile:

Comment (aboguszewski-sumo) [reply tests/integration/internal/constants.go:333]: This PR is WIP, sorry 😞

File: tests/integration/internal/constants.go [hunk 2]
```diff
@@ -392,3 +412,26 @@ func InitializeConstants() error {
...
+	require.NoError(t, err)
+	return v
+}
+
+func GetVersionDependentMetrics(t *testing.T) []string {
+	res := []string{}
+	currVersion := getKubernetesVersion(t)
+
+	for version, ms := range versionDependentMetrics {
+		if currVersion < version {
```
Comment (swiatekm-sumo) [inline tests/integration/internal/constants.go:429]: This is a string comparison, right? If we want to compare this way, we should use a more structured data type for this.

Comment (swiatekm-sumo) [reply tests/integration/internal/constants.go:433]: https://pkg.go.dev/k8s.io/apimachinery/pkg/util/version looks like a good choice.

**Review by swiatekm-sumo — APPROVED** [review]

**Pull request merged**